app/src/widgets/MessageList/ui/MessageList.tsx

```diff
@@ -1,24 +1,85 @@
-import { ChangeEvent, FC, useCallback, useEffect, useState } from 'react';
+import {
+    ChangeEvent,
+    FC,
+    useCallback,
+    useEffect,
+    useRef,
+    useState,
+} from 'react';
 import SearchOutlined from '@ant-design/icons/SearchOutlined';
 
 import { Card, Input } from 'ui-kit';
 import { InputVariants } from 'ui-kit/Input';
 
 import { getAllCompactMessages } from '@entities/compactMessage/api/getAllCompactMessages';
+import { searchCompactMessages } from '@entities/compactMessage/api/searchCompactMessages';
 import { CompactMessage } from '@entities/compactMessage/model/compactMessage';
 
+import { useDebounce } from '@shared/lib/hooks/useDebounce';
+import { useInfiniteScroll } from '@shared/lib/hooks/useInfiniteScroll';
+import { useIsToggle } from '@shared/lib/hooks/useIsToggle';
+
 import css from './MessageList.module.scss';
 
 const MessageList: FC = () => {
     const [valueInput, setValueInput] = useState<string>('');
     const [compactMessages, setCompactMessages] = useState<CompactMessage[]>(
         [],
     );
+    const [page, setPage] = useState<number>(1);
+
+    const rootElement = useRef<HTMLDivElement | null>(null);
+
+    const valueDebounce = useDebounce<string>(valueInput, 500);
+
+    const { toggle, handleChangeToggle } = useIsToggle();
```

Comment:
если происходит надобность из-за линтера  то проще уж тут
```suggestion
    const { isToggled, toggle: handleChangeToggle } = useIsToggled();
```